applications/serial_lte_modem/src/slm_at_host.c

```diff
@@ -782,9 +767,9 @@ int slm_at_host_init(void)
 		return -EFAULT;
 	}
 
-	err = at_notif_register_handler(NULL, response_handler);
+	err = nrf_modem_at_notif_handler_set(notification_handler);
```

Comment:
You shouldn't register your handler like so, but instead declare an AT monitor to receive the notifications like described here: https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/libraries/modem/at_monitor.html?highlight=monitor#wildcard-filter

When you use `nrf_modem_at_notif_handler_set`, you send all notifications to this handler, and AT monitors won't work. Instead, register a monitor and let AT monitor register its handler function with libmodem, which dispatches incoming notifications to all monitors.

Reply:
Also, once you have defined the `notification_handler` as an AT monitor it will be already executed by the system workqueue so you won't need a dedicated work to offload.

Reply:
Thanks, I already defined a AT_MONITOR for GNSS service
https://github.com/nrfconnect/sdk-nrf/blob/master/applications/serial_lte_modem/src/gnss/slm_at_gnss.c#L153

If I define another ANY monitor, will they interfere to each other? For example, the NCELLMEAS notification in GNSS service should NOT be captured by the ANY monitor, which will in turn send to MCU side.

Reply:
A monitor with filter `ANY` will receive all notifications, but you can ignore `NCELLMEAS` notifications in the handler itself, for example by returning early if the notifications contains the substring `NCELLMEAS`.

Reply:
This is difficult. If MCU issues `%NCELLMEAS`, the notifications should not be ignored; when SLM issues `%NCELLMEAS`, it should be ignored. I'll have to modify GNSS service not to issue `%NCELLMEAS` inside SLM, but request MCU to issue it, then notification captured by two AT_MONITORs.

Reply:
@lemrey updated as described, tested OK, please review again.